### PR TITLE
[PaddlePaddle Hackathon] Task 71: Mask-RCNN compression

### PR DESCRIPTION
**Pull request by gbstack**
**PR types**

New features

**PR changes**

APIs

**Describe**

Hi,

This PR add pruning for Mask-RCNN.

The new pruning configuration file is located at _configs/slim/prune/mask_rcnn_r50_prune_fpgm.yml_.

The model parameters count ratio after pruning is 0.8.

For improving training, 5 epoches (learning rate will be decreased by 10 times at epoch 3) are used and learning rate is 1/10 of original learning rate. And weight decay is set to 0 according to this post: https://docs.nvidia.com/tao/tao-toolkit/text/instance_segmentation/mask_rcnn.html#pruning-the-model.

Test result at epoch 3

    Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.320
    Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.506
    Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.338
    Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.171
    Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.364
    Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.417
    Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.291
    Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.459
    Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.479
    Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.272
    Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.537
    Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.620

### CODE REVIEW

**Comment by lyuwenyu** [comment]
这个有最后的结果嘛

**Comment by jerrywgz** [comment]
在https://github.com/PaddlePaddle/PaddleDetection/tree/develop/configs/slim 文档里补充下最终模型训练精度和速度对比吧

**Comment by gbstack** [comment]
> 这个有最后的结果嘛

目前训练到了epoch 3

**Comment by gbstack** [comment]
> 在https://github.com/PaddlePaddle/PaddleDetection/tree/develop/configs/slim 文档里补充下最终模型训练精度和速度对比吧

好的

**Comment by lyuwenyu** [comment]
> 目前训练到了epoch 3

要不训练完吧

**Comment by gbstack** [comment]
> > 目前训练到了epoch 3
> 
> 要不训练完吧

好的，正在训练

**Comment by gbstack** [comment]
最后的训练结果是这样的

epoch 4

Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.320
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.505
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.338
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.171
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.364
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.417
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.290
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.459
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.479
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.272
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.537
Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.620
